src/Chapter3.hs

```diff
@@ -480,6 +546,81 @@ After defining the city, implement the following functions:
    and at least 10 living __people__ inside in all houses of the city in total.
 -}
 
+data Person 
+    = Person 
+    deriving (Show)
+
+data House = House 
+    { houseResidents :: [Person]
```

Comment:
Interesting representation. However, this doesn't guarantee that the instance of this data type will be correct. Your data type allows the house to be without a person, or with 1000 persons. Also, the `Person` data type itself is redundant as it is just an empty type. You can notice that it is more convenient to have a notion of "Number of Persons" in this type. And to restrict the number of persons, you can create 4 constructors for each option of houses.